src/com/owncloud/android/ui/activity/ReceiveExternalFilesActivity.java

```diff
@@ -398,11 +404,15 @@ private void populateDirectoryList() {
         if (mFile != null) {
             // TODO Enable when "On Device" is recovered ?
             Vector<OCFile> files = getStorageManager().getFolderContent(mFile/*, false*/);
-            List<HashMap<String, OCFile>> data = new LinkedList<>();
+            files = sortFileList(files);
+
+            List<HashMap<String, Object>> data = new LinkedList<>();
             for (OCFile f : files) {
-                HashMap<String, OCFile> h = new HashMap<>();
+                if (f.isFolder()) {
```

Comment:
No, it was requested. Some users were confused because they couldn't see if the could overwrite a file already existing with the same name, or the file was already uploaded.